src/openmmio.py

```diff
@@ -1117,34 +1168,68 @@ def normal_modes(self, shot=0, optimize=True):
         normal_modes = normal_modes[larger_freq_idxs]
         return freqs, normal_modes
 
-    def optimize(self, shot=0, crit=1e-4):
+    def optimize(self, shot, crit=1e-4, disable_vsite=False, align=True, include_restraint_energy=False):
+
+        """ 
+        Optimize the geometry and align the optimized 
+        geometry to the starting geometry.
+        
+        Parameters
+        ----------
+        shot : int
+            The snapshot number to be minimized
+        crit : float
+            Convergence criterion in kJ/mol
+        disable_vsite : bool
+            Disable virtual sites (needed for SMIRNOFF)
+        include_restraint_energy : bool
+            Include energy component from CustomExternalForce
 
-        """ Optimize the geometry and align the optimized geometry to the starting geometry, and return the RMSD. """
+        Returns
+        -------
+        E : float
+            Potential energy of the system
+        rmsd : float
+            RMSD of the system (w/r.t. starting geometry) in Angstrom
+        """
 
         steps = int(max(1, -1*np.log10(crit)))
         self.update_simulation()
         self.set_positions(shot)
+        if hasattr(self, 'restrain_k'):
```

Comment:
I guess it's possible that `self.restrain_k` will get a default value of `0.0`. In this case we should skip this so maybe it's better to write
```
if hasattr(self, 'restrain_k') and self.restrain_k != 0:
```

Reply:
Good point. I will make the replacement everywhere I check for this attribute.